code/class/registry/registry.php

```diff
@@ -129,7 +136,54 @@ public function alias($class, $alias)
     }
 
     /**
-     * Get a list of all the identifier aliases
+     * Get a specific locator from the class namespace
+     *
+     * @return string|false The name of the locator or FALSE if no locator could be found
+     */
+    public function getLocator($class)
+    {
+        $result = false;
+
+        if(!false == $pos = strrpos($class, '\\'))
+        {
+            $namespace = substr($class, 0, $pos);
+
+            if(isset($this->_namespaces[$namespace])) {
+                $result = $this->_namespaces[$namespace];
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * Set a specific locator based on a class namespace
+     *
+     * @return  ClassRegistry
+     */
+    public function setLocator($class, $locator)
+    {
+        if(!false == $pos = strrpos($class, '\\'))
```

Comment:
Just put together a test for this, and it confirms my suspicions. Once a class is loaded with a locator for a certain namespace, no other locators registered to that same namespace will ever be used.

```php
/**
 * Tests registering a locator with the class loader & getting the registered locator
 */
public function testRegisterMultipleLocatorSameNamespace()
{
    $this->_loader->registerLocator(new ClassLocatorFixture([
        'namespaces' => [
            'Fixture' => $this->_basedir . '/fixture'
        ]
    ]));
    
    $this->_loader->registerLocator(new ClassLocatorStub([
        'namespaces' => [
            'Fixture' => $this->_basedir . '/stub'
        ]
    ]));

    $this->assertEquals($this->_basedir . '/fixture/stub.php', $this->_loader->getPath('Fixture\\Stub'));
}
```
```
1) ClassLoaderTest::testRegisterMultipleLocatorSameNamespace
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'/Users/oli/Projects/kodekit/tests/fixtures/classes/fixture/stub.php'
+'/Users/oli/Projects/kodekit/tests/fixtures/classes/fixture/loader.php'

/Users/oli/Projects/kodekit/tests/unit/class/loader/ClassLoaderTest.php:139
```